test/extended/setup.sh

```diff
@@ -11,17 +11,25 @@ function os::test::extended::focus () {
 	if [[ -n "${FOCUS:-}" ]]; then
 		exitstatus=0
 
+		local excludesFile="${ARTIFACT_DIR}/parallel_test_names.txt"
+
 		# first run anything that isn't explicitly declared [Serial], and matches the $FOCUS, in a parallel mode.
 		os::log::info "Running parallel tests N=${PARALLEL_NODES:-<default>} with focus ${FOCUS}"
-		TEST_REPORT_FILE_NAME=focus_parallel TEST_PARALLEL="${PARALLEL_NODES:-5}" os::test::extended::run -- -ginkgo.skip "\[Serial\]" -test.timeout 6h ${TEST_EXTENDED_ARGS-} || exitstatus=$?
+		TEST_REPORT_FILE_NAME=focus_parallel TEST_PARALLEL="${PARALLEL_NODES:-5}" \
+			os::test::extended::run -- \
+				-print-test-names-to "${excludesFile}" \
+				-ginkgo.skip "\[Serial\]" \
+				-test.timeout 6h ${TEST_EXTENDED_ARGS-} || exitstatus=$?
 
 		# Then run everything that requires serial and matches the $FOCUS, serially.
 		# there is bit of overlap here because not all serial tests declare [Serial], so they might have run in the
 		# parallel section above.  Hopefully your focus was precise enough to exclude them, and we should be adding
 		# the [Serial] tag to them as needed.
```

Comment:
since you're fixing this, please clean up the comments related to the potential for tests to be run twice.

Reply:
Cleaned up.